qemu.go

```diff
@@ -217,32 +217,46 @@ func (q *qemu) appendFSDevices(devices []ciaoQemu.Device, podConfig PodConfig) [
 	return devices
 }
 
-func (q *qemu) appendConsoles(devices []ciaoQemu.Device, podConfig PodConfig) []ciaoQemu.Device {
+func (q *qemu) appendConsoles(devices []ciaoQemu.Device, podConfig *PodConfig) []ciaoQemu.Device {
 	serial := ciaoQemu.SerialDevice{
 		Driver: ciaoQemu.VirtioSerial,
 		ID:     "serial0",
 	}
 
 	devices = append(devices, serial)
 
+	if podConfig.Console == "" {
+		podConfig.Console = fmt.Sprintf("%s/%s/console.sock", runStoragePath, podConfig.ID)
+	}
+
+	console := ciaoQemu.CharDevice{
+		Driver:   ciaoQemu.Console,
+		Backend:  ciaoQemu.Socket,
+		DeviceID: "console0",
+		ID:       "charconsole0",
+		Path:     podConfig.Console,
+	}
+
+	devices = append(devices, console)
+
 	for i, c := range podConfig.Containers {
-		var console ciaoQemu.CharDevice
-		if c.Interactive == false || c.Console == "" {
-			consolePath := fmt.Sprintf("%s/%s/console.sock", runStoragePath, podConfig.ID)
+		// Need to add an offset because of the console created for the pod.
+		idx := i + 1
 
+		if c.Interactive == false || c.Console == "" {
 			console = ciaoQemu.CharDevice{
 				Driver:   ciaoQemu.Console,
 				Backend:  ciaoQemu.Socket,
-				DeviceID: fmt.Sprintf("console%d", i),
-				ID:       fmt.Sprintf("charconsole%d", i),
-				Path:     consolePath,
+				DeviceID: fmt.Sprintf("console%d", idx),
+				ID:       fmt.Sprintf("charconsole%d", idx),
+				Path:     fmt.Sprintf("%s/%s/%s/console.sock", runStoragePath, podConfig.ID, c.ID),
```

Comment:
@mcastelino not sure I understand what you suggest here. We can add something to the runtime, but this is not part of virtcontainers. This patch is intended to enable something which was not working until now.